bundles/org.openhab.binding.mycroft/src/main/java/org/openhab/binding/mycroft/internal/channels/FullMessageChannel.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mycroft.internal.channels;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.mycroft.internal.MycroftBindingConstants;
+import org.openhab.binding.mycroft.internal.MycroftHandler;
+import org.openhab.binding.mycroft.internal.api.MessageType;
+import org.openhab.binding.mycroft.internal.api.dto.BaseMessage;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.Command;
+
+/**
+ * The channel responsible for sending/receiving raw message
+ *
+ * @author Gwendal ROULLEAU - Initial contribution
+ */
+@NonNullByDefault
+public class FullMessageChannel extends MycroftChannel<StringType> {
+
+    private List<String> messageTypesList = new ArrayList<>();
+
+    public FullMessageChannel(MycroftHandler handler, String messageTypesList) {
+        super(handler, MycroftBindingConstants.FULL_MESSAGE_CHANNEL);
+        for (String messageType : messageTypesList.split(",")) {
+            this.messageTypesList.add(messageType.trim());
+        }
+    }
+
+    @Override
+    public List<MessageType> getMessageToListenTo() {
+        return Arrays.asList(MessageType.any);
+    }
+
+    @Override
+    public void messageReceived(BaseMessage message) {
+        if (messageTypesList.contains(message.type.getMessageTypeName())) {
+            updateMyState(new StringType(message.message));
+        }
+    }
+
+    @Override
+    public void handleCommand(Command command) {
+        if (command instanceof StringType) {
+            if (handler.sendMessage(command.toString())) {
```

Comment:
I am not 100% sure but you might want to use `toFullString()` here if you want to have the **full* content of that command/state. The `toString()` methods for openHAB objects are designed for writing them to the console or log files, whereas the `toFullString()` are defined as printing out everything that is necessary to construct the correct command/state object again.

Reply:
OK, seems to work, done